src/DB.php

```diff
@@ -660,6 +660,12 @@ public static function insertMessageRequest(Message &$message)
         $chat_id = $chat->getId();
 
         $date = self::getTimestamp($message->getDate());
+        $edit_date = self::getTimestamp($message->getEditDate());
+
+        if ($edit_date == $date) {
```

Comment:
This condition will (pretty much) never be true, because `edit_date` is never set at this point. So `$edit_date` will always have the current timestamp (because `null` is being passed to `getTimestamp()`).
Hence, `$date` is always a bit earlier than `$edit_date`.

What you'd want is something like this:
```php
$date = self::getTimestamp($message->getDate());
if ($edit_date = $message->getEditDate()) {
    $edit_date = self::getTimestamp($edit_date);
    if ($date === $edit_date) {
        $edit_date = null;
    }
}
```

But like I said, this always fails, because if a message is being edited, `insertEditedMessageRequest()` is called, not `insertMessageRequest()`

Please correct me if I'm wrong!